include/seqan3/range/view/translate_join.hpp

```diff
@@ -0,0 +1,390 @@
+        requires ConstIterableRange<urng_t>
+    {
+        assert(n < size());
+        size_type index_frame = n % selected_frames.size();
+        size_type index_urange = (n - index_frame) / selected_frames.size();
+        return urange[index_urange] | view::translate_single(selected_frames[index_frame]);
+    }
+    //!\}
+};
+
+//!\brief Class template argument deduction for view_translate_join.
+template <typename urng_t>
+//!\cond
+    requires std::ranges::SizedRange<urng_t> &&
+             std::ranges::RandomAccessRange<urng_t> &&
+             std::ranges::SizedRange<reference_t<urng_t>> &&
+             std::ranges::RandomAccessRange<reference_t<urng_t>> &&
+             NucleotideAlphabet<reference_t<reference_t<urng_t>>>
+//!\endcond
+view_translate_join(urng_t &&, translation_frames const = translation_frames{}) -> view_translate_join<std::ranges::all_view<urng_t>>;
+
+// ============================================================================
+//  translate_fn (adaptor definition for both views)
+// ============================================================================
+
+//!\brief Definition of the range adaptor object type for seqan3::view::translate_join.
+struct translate_join_fn
+{
+    //!\brief Store the argument and return a range adaptor closure object.
+    constexpr auto operator()(translation_frames const tf = translation_frames::SIX_FRAME) const
+    {
+        return detail::adaptor_from_functor{*this, tf};
+    }
+
+    /*!\brief            Directly return an instance of the view, initialised with the given parameters.
+     * \param[in] urange The underlying range.
+     * \param[in] tf     The frame that should be used for translation.
+     * \returns          A range of translated sequence(s).
+     */
+    template <std::ranges::Range urng_t>
+    constexpr auto operator()(urng_t && urange, translation_frames const tf = translation_frames::SIX_FRAME) const
+    {
+        static_assert(dimension_v<urng_t> == 2,
+            "This adaptor only handles range-of-range (two dimensions) as input.");
+        static_assert(std::ranges::ViewableRange<urng_t>,
+            "The range parameter to view::translate_join cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::ViewableRange<urng_t>,
+            "The inner range of the range parameter to view::translate_join cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::SizedRange<urng_t>,
+            "The range parameter to view::translate_join must model std::ranges::SizedRange.");
+        static_assert(std::ranges::SizedRange<urng_t>,
+            "The inner range of the range parameter to view::translate_join must model std::ranges::SizedRange.");
+        static_assert(std::ranges::RandomAccessRange<urng_t>,
+            "The range parameter to view::translate_join must model std::ranges::RandomAccessRange.");
+        static_assert(std::ranges::RandomAccessRange<urng_t>,
```

Comment:
You forgot `reference_t` for each of the copy'n'pasted concept checks!

Reply:
Classic! Sorry, I fixed it ..